src/ipz_parser.cpp

```diff
@@ -631,4 +631,117 @@ types::DbusVariantType IpzVpdParser::readKeywordFromHardware(
+
+bool IpzVpdParser::isRecordECCValid(const auto& i_recordDataOffset,
+                                    const auto& i_recordDataLength,
+                                    const auto& i_recordECCOffset,
+                                    const auto& i_recordECCLength)
+{
+    (void)i_recordDataOffset;
+    (void)i_recordDataLength;
+    (void)i_recordECCOffset;
+    (void)i_recordECCLength;
+
+    return false;
+}
+
+int IpzVpdParser::setKeywordValueInRecord(
+    const types::Record& i_recordName, const types::Keyword& i_keywordName,
+    const types::BinaryVector& i_keywordData,
+    const types::RecordOffset& i_recordDataOffset)
+{
+    (void)i_recordName;
+    (void)i_keywordName;
+    (void)i_keywordData;
+    (void)i_recordDataOffset;
+
+    return -1;
+}
+
+int IpzVpdParser::writeKeywordValueOnHardware(
+    const types::WriteVpdParams i_paramsToWriteData)
+{
+    // Extract record, keyword and value from i_paramsToWriteData
+    types::Record l_recordName;
+    types::Keyword l_keywordName;
+    types::BinaryVector l_keywordData;
+
+    if (const types::IpzData* l_ipzData =
+            std::get_if<types::IpzData>(&i_paramsToWriteData))
+    {
+        l_recordName = std::get<0>(*l_ipzData);
+        l_keywordName = std::get<1>(*l_ipzData);
+        l_keywordData = std::get<2>(*l_ipzData);
+    }
+    else
+    {
+        logging::logMessage(
+            "Input parameter type provided isn't compatible with the given VPD type.");
+        throw types::DbusInvalidArgument();
+    }
+
+    if (l_recordName == "VHDR" || l_recordName == "VTOC")
+    {
+        logging::logMessage("Write operation not allowed on the given record.");
+
+        // Zero bytes updated.
+        return 0;
+    }
+
+    if (l_keywordData.size() == 0)
+    {
+        logging::logMessage("Given keyword's data length is 0.");
+
+        // Zero bytes updated.
+        return 0;
+    }
+
+    auto l_vpdBegin = m_vpdVector.cbegin();
+
+    // Get VTOC offset
+    std::ranges::advance(l_vpdBegin, Offset::VTOC_PTR, m_vpdVector.cend());
+    auto l_vtocOffset = readUInt16LE(l_vpdBegin);
+
+    // Get the details of user given record from VTOC
+    const types::RecordData& l_inputRecordDetails =
+        getRecordDetailsFromVTOC(l_recordName, l_vtocOffset);
+
+    const auto& l_inputRecordOffset = std::get<0>(l_inputRecordDetails);
+    const auto& l_inputRecordLength = std::get<1>(l_inputRecordDetails);
+    const auto& l_inputRecordECCOffset = std::get<2>(l_inputRecordDetails);
+    auto l_inputRecordECCLength = std::get<3>(l_inputRecordDetails);
+
+    // Validate the user given record's ECC
+    if (!isRecordECCValid(l_inputRecordOffset, l_inputRecordLength,
+                          l_inputRecordECCOffset, l_inputRecordECCLength))
+    {
+        logging::logMessage(
+            "Given record's ECC is not valid. Aborting write operation.");
+        return -1;
+    }
+
+    // write keyword's value on hardware
+    int l_sizeWritten = setKeywordValueInRecord(
```

Comment:
What is the API fails and -1 is returned?

Reply:
add the logic, to check for sizeWritten - before updating the ECC